### PR TITLE
Plots: Add the bounding boxes informations in the `dvc plots diff` command so that VScode can display them.

### LINKED ISSUE #10198: plots: interactive plots with toggling bounding box
Issue stems from:
https://github.com/iterative/vscode-dvc/issues/4917

The idea is to have an interactive plot (likely not a template of the current plot system), for which the user can view different labels for each class, and toggle labels on and off to see bounding boxes for specific classes.

### PR DESCRIPTION
**Pull request by AlexandreKempf**
Works with [this DVClive PR](https://github.com/iterative/dvclive/pull/776) and [this VSCode PR](https://github.com/iterative/vscode-dvc/pull/5227).

DVClive is responsible for saving JSON annotation files. The annotations files have the same path as the image they match, but ends with `.json`. 

Example:
```
plots/images/dogs.png
plots/images/dogs.json
```
Or for `step` images:
```
plots/images/cats/0.png
plots/images/cats/0.json
plots/images/cats/1.png
plots/images/cats/1.json
```

The JSON files have this format:
```json
{
    "annotations": {
        "person": [
            {
                "box": {
                    "top": 1,
                    "left": 10,
                    "bottom": 30,
                    "right": 40
                },
                "score": 0.8
            }
        ],
        "dog": [
            {
                "box": {
                    "top": 200,
                    "left": 50,
                    "bottom": 240,
                    "right": 160
                },
                "score": 0.3
            },
            {
                "box": {
                    "top": 57,
                    "left": 34,
                    "bottom": 246,
                    "right": 144
                },
                "score": 0.2
            }
        ]
    }
}
```

After the `dvc plots diff --split --json` the expected output is to have the annotations alongside the image
```json
{
  "data": {
    "plots/bounding_boxes.png": [
      {
        "type": "image",
        "revisions": [
          "workspace"
        ],
        "url": "//bounding_boxes.png",
        "annotations": {
          ... content from the json file "annotations" field should be here
        }
      }
    ]
  }
}
```

:warning: Mising unit tests, I might need help on this. Tests in this repo are complicated to navigate on a first PR.

Fixes #10198

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [ ] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: dvc/repo/plots/__init__.py [hunk 1]
```diff
@@ -357,6 +358,10 @@ def infer_data_sources(plot_id, config=None):
     if isinstance(x, dict):
         sources.append(first(x.keys()))
 
+    # annotation = config.get("annotations", None)
+    # if isinstance(annotation, str):
+    #     sources.append(annotation)
+
```
Comment (dberenbaum) [inline dvc/repo/plots/__init__.py:364]: Does this need to get dropped or uncommented?

File: dvc/render/converter/image.py [hunk 2]
```diff
@@ -46,15 +47,25 @@ def flat_datapoints(self, revision: str) -> tuple[list[dict], dict]:
         path = self.properties.get("out")
         datapoints = []
         datas, properties = self.convert()
-        for filename, _, image_data in datas:
+
+        if "annotations" in properties:
+            with open(properties["annotations"], encoding="utf-8") as annotations_path:
+                annotations = json.load(annotations_path)
```
Comment (dberenbaum) [inline dvc/render/converter/image.py:53]: What if there is a json file that doesn't conform to the expected format?

File: dvc/repo/plots/__init__.py [hunk 3]
```diff
@@ -459,6 +465,28 @@ def _resolve_definitions(
...
+    for path in target["data"]:
+        annotation_file = Path(path).with_suffix(".json").as_posix()
+        if ImageRenderer.matches(path) and annotation_file in target["data"]:
+            annotations = {"annotations": annotation_file}
+            # empty dict all share the same reference, so override them
+            if target["data"][path]:
+                target["data"][path].update({"annotations": annotation_file})
+            else:
+                target["data"][path] = annotations
+            path_to_remove.append(annotation_file)
```
Comment (dberenbaum) [inline dvc/repo/plots/__init__.py:482]: Can you explain a bit what's happening here?

**Comment by mattseddon** [comment]
Needs tests

File: dvc/repo/plots/__init__.py (hunk 1 above)
Comment (AlexandreKempf) [reply dvc/repo/plots/__init__.py:364]: I kept it because Studio needs it to work :)

File: dvc/repo/plots/__init__.py (hunk 3 above)
Comment (AlexandreKempf) [reply dvc/repo/plots/__init__.py:482]: Let me know if it is clear now ;)

File: dvc/repo/plots/__init__.py [hunk 4]
```diff
@@ -459,6 +465,31 @@ def _resolve_definitions(
...
+    for path in target["data"]:
+        annotation_file = Path(path).with_suffix(".json").as_posix()
+        # if an image and a json match names, we consider the JSON to be annotations
+        # in that case, we add the annotations to the image "definition" as a property
+        # so they are link together.
+        if ImageRenderer.matches(path) and annotation_file in target["data"]:
+            # target["data"] empty dict all share the same reference, so override them
+            annotations = {"annotations": annotation_file}
+            if target["data"][path]:
+                target["data"][path].update({"annotations": annotation_file})
```
Comment (dberenbaum) [inline dvc/repo/plots/__init__.py:482]:
```suggestion
                target["data"][path].update(annotations)
```

**Comment by skshetry** [comment]
Hey @mattseddon. Could you please take a look at this PR (and maybe takeover)?

**Comment by mattseddon** [comment]
will get back to this later

**Pull request closed without merging**